packages/firebase/index.d.ts

```diff
@@ -7087,6 +7087,13 @@ declare namespace firebase.firestore {
    */
   export function setLogLevel(logLevel: LogLevel): void;
 
+  export interface FirestoreDataConverter<T> {
+    // Converts a model object of type T into plain Firestore DocumentData.
+    toFirestore(modelObject: T): DocumentData;
+    // Converts a Firestore DocumentSnapshot into a model object of type T.
+    fromFirestore(snapshot: DocumentSnapshot, options: SnapshotOptions): T;
+  }
+
   /**
```

Comment:
Wrote a basic checker for the sake of this, but I'll definitely explore an automated script that checks this. It's a little tricky right now because some names are different and some one-liner functions are spread out across multiple lines in firestore-types, but I will bother Gil and learn how to script.